docs/applications/containers/how-to-deploy-the-elastic-stack-on-kubernetes/index.md

```diff
@@ -0,0 +1,293 @@
+
+        helm install --name elasticsearch --wait --timeout=600 elastic/elasticsearch
+
+1.  A three-node Elasticsearch cluster is now configured and available locally to the Kubernetes cluster. To confirm this, first port-forward a local port to the Elasticsearch service. You should leave this command running in a terminal window or tab in in the background for the remainder of this tutorial.
+
+        kubectl port-forward svc/elasticsearch-master 9200:9200
+
+1.  In another terminal window, send a request to this port:
+
+        curl http://localhost:9200/
+
+1.  You should see a response similar to the following:
+
+        {
+          "name" : "elasticsearch-master-0",
+          "cluster_name" : "elasticsearch",
+          "cluster_uuid" : "o66WYOm5To2znbZ0kOkDUw",
+          "version" : {
+            "number" : "7.1.1",
+            "build_flavor" : "default",
+            "build_type" : "docker",
+            "build_hash" : "7a013de",
+            "build_date" : "2019-05-23T14:04:00.380842Z",
+            "build_snapshot" : false,
+            "lucene_version" : "8.0.0",
+            "minimum_wire_compatibility_version" : "6.8.0",
+            "minimum_index_compatibility_version" : "6.0.0-beta1"
+          },
+          "tagline" : "You Know, for Search"
+        }
+
+Note that your specific version numbers and dates may be different in this json response. Elasticsearch is operational, but not receiving or serving any data.
+
+### Install Filebeat
+
+In order to start processing data, deploy the `filebeat` chart to your Kubernetes cluster. This will collect all pod logs and store them in Elasticsearch, after which they can be searched and used in visualizations within Kibana.
+
+1.  Deploy the `filebeat` chart. No custom `values.yaml` file should be necessary:
+
+        helm install --name filebeat --wait --timeout=600 elastic/filebeat
+
+1.  Confirm that Filebeat has started to index documents into Elasticsearch by sending a request to the locally-forwarded Elasticsearch service port:
+
+        curl http://localhost:9200/_cat/indices
+
+1.  At least one `filebeat` index should be present, similar to the following:
+
+        green open filebeat-7.1.1-2019.06.25-000001 _7Rw8LkvTeKpJPly7cpzNw 1 1 9886 0 5.7mb 2.8mb
+
+### Install Kibana
+
+Kibana will provide a frontend to Elasticsearch and the data collected by Filebeat.
+
+1.  Deploy the `kibana` chart:
+
+        helm install --name kibana --wait --timeout=600 elastic/kibana
+
+1.  Port-forward the `kibana-kibana` service in order to access Kibana locally. Leave this command running in the background as well for the remainder of this tutorial.
+
+        kubectl port-forward svc/kibana-kibana 5601:5601
+
+## Configure Kibana
+
+Before visualizing pod logs, Kibana must be configured with an index pattern for Filebeat's indices.
+
+1.  With the previous `port-forward` command running in another terminal window, open your browser and navigate to http://localhost:5601
```

Comment:
If you're running this on, say a Linode, wouldn't you be able to replace localhost with the IP address of that Linode to go to this page from another computer? I cannot get this to work. Do I need to open some other port? Or should we instead note that this should all be run on a local computer with a web bowser available and not a Linode?

Reply:
The challenge behind accessing a _kubernetes_ service is that unless a load-balancer aware external ingress is present or the service is configured in such a way (`NodePort`) to simply be exposed as a port on the underlying Linode instance, services tend to be private by default. In this case, that route is generally the best approach: unless the user has more experience with the service in question and knows how to secure and operate it properly, keeping it private within kubernetes and only specifically exposed is a safer option.

Do you encounter any errors with the `kubectl port-forward svc/kibana-kibana 5601:5601` command? Alternatively, are you running a particularly aggressive host-based firewall that would limit your ability to hit a local listener like `:5601`?

Reply:
Hi Tyler - I didn't have any trouble with that command and don't have any firewall installed, but still have trouble. I tried pointing a domain name to various nodes in the cluster as well as my Linode that created the cluster as well to no success. I'll do this again setting it up on my local machine instead. And keep you posted. Thanks for your help!